src/ClusterTSnePlot.js

```diff
@@ -97,9 +98,59 @@ const ClusterTSnePlot = (props) => {
     <option key={perplexity} value={perplexity}>{perplexity}</option>
   ))
 
-  const kOptions = ks.sort((a, b) => a - b).map((k) => (
-    <option key={k} value={k}>{k}</option>
-  ))
+  const kOptions = ks.sort((a, b) => a-b).map((k) => ({
+    value: k,
+    label: k,
+    group: 'clusters'
+  }))
+
+  const metadataOptions = metadata.map((metadata) => ({
+    ...metadata,
+    group: "metadata"
+  }))
+
+  const options = [
+    {
+      label: 'Metadata',
+      options: metadataOptions,
+    },
+    {
+      label: 'Number of clusters',
+      options: kOptions,
+    },
+  ]
+
+  const formatGroupLabel = data => (
```

Comment:
Just because you can, it doesn't mean you should 😁 I do find it a bit more readable with the added brackets, but maybe I'm too old school and I need to get with the modern times.